git/git.go

```diff
@@ -311,7 +311,10 @@ func (provider *repositoryProvider) snapshot(repository *git.Repository, commit
 
 	tree, err := commit.Tree()
 	if err != nil {
-		return 0, fmt.Errorf("failed to get tree of commit '%v': %v", commit.Hash, err)
+        return 0, util.ErrorWithCode{
```

Comment:
jw, why is the return code 0 if there is an error?

Reply:
The return code isn't 0, that's the count being returned.
The actual return code is in the `StatusCode` field.